Add "Next.js with Chakra UI" template

spykr opened the pull request:
This template contains the following:

- Next.js, with:
  - TypeScript support
  - Custom `_app.tsx`, `_document.tsx` and `_error.tsx`
  - Example `index.tsx` and `404.tsx` pages
  - Example `hello.ts` API route
  - Absolute imports (e.g. `import Button from '@/components/Button'`)
  - Example meta tags and favicon
  - `next-optimized-images` plugin pre-installed
- Chakra-UI, with example custom theme
- ESLint & Prettier, with VSCode integration
- Sentry integration for both front-end pages and API routes (simply add env variables)
- Google Analytics integration (simply add env variables)
- Jest with React-Testing-Library (and example tests)
- Cypress with Cypress-Testing-Library (and example tests)
- Storybook (with addons and example stories)
  - Storycap for taking screenshots of pages on various browser sizes
- Google Web Font included as an example

spykr committed:
Add "Next.js with Chakra UI" template

andrewfulton commented on template/next-ts-chakra-ui/src/components/Provider.tsx:
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
```
[not sure how to add file-level comment]. Is `Provider` too general a name for this? Feels like we may as well have `Component.tsx`

spykr replied:
I'm not too sure what you mean by this. Are you saying you think it should be more specific, like `GlobalProvider` or `AppProvider`?

andrewfulton replied:
Thats what I'm getting at yeah - it's basically a ThemeProvider or StyleProvider at the moment right? 
Is the pattern that it would begin to provide multiple things? Or we would have
```
<Multiple>
  <LevelsOfProviders>
    <WrappingThe>
       <PageComponent />
```

spykr replied:
The idea was that the one component would provide everything, since it needs to be used in 3 places (`_app.tsx`, Storybook and Jest) I'd rather not have to change all 3 places because I added Redux or what have you.

spykr replied:
I've renamed it to `AppProvider` 👍

andrewfulton commented on template/next-ts-chakra-ui/pages/index.tsx:
```diff
@@ -0,0 +1,17 @@
+import { Home } from '@/components/pages';
+import generateMetaTags from '@/utils/meta';
+
+const Index: NextPage = () => {
```
If we are going to use this as our Best Practice Broilerplate, it would be good to probably have a getServerSideProps() typed function in here.

spykr replied:
Good point, this slipped my mind. Thanks 👍 
I guess I'll add a clock that shows the current time or something 😛

spykr replied:
Done 👍

spykr committed:
Add current time to home page to demonstrate `getServerSideProps` function

andrewfulton commented on template/next-ts-chakra-ui/pages/index.tsx:
```diff
@@ -1,17 +1,28 @@
   );
 };
 
+export const getServerSideProps: GetServerSideProps = async () => {
```
should this be typed too? `GetServerSideProps<Props>` or what have you?

spykr replied:
Ah, didn't know you could do that. Fixed.

spykr committed:
Add props to `getServerSideProps` type

spykr committed:
Change default button width to `auto`

spykr committed:
Rename Provider to AppProvider

spykr committed:
Update .gitignore

spykr committed:
Update README

spykr committed:
Update root README and package.json

spykr committed:
Adjust Jest config to handle more file types

spykr committed:
Update dependencies

- Added MockRouterProvider for Jest and Storybook since in Next 9.5 the Link component will throw an error if no router context is provided

spykr committed:
Fix console errors

spykr committed:
Adjust to output package.json keys in a prettier order

spykr committed:
Update Storybook to attach `withKnobs` decorator globally

theleebriggs reviewed (approved):
Lovely stuff @spykr . I think anything else we'd want will come out in further use - great to have this solid base.

spykr committed:
Update "Learn more" section of README
